components/ui/sidebar.tsx

```
@@ -1,763 +1,763 @@
-          {
-            "--skeleton-width": width,
-          } as React.CSSProperties
-        }
-      />
-    </div>
-  )
-})
-SidebarMenuSkeleton.displayName = "SidebarMenuSkeleton"
+	// Random width between 50 to 90%.
+	const width = React.useMemo(() => {
+		return `${Math.floor(Math.random() * 40) + 50}%`;
```

ESLint

[error] Error: Cannot call impure function during render

`Math.random` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

/home/jailuser/git/components/ui/sidebar.tsx:655:24
  653 | 	// Random width between 50 to 90%.
  654 | 	const width = React.useMemo(() => {
> 655 | 		return `${Math.floor(Math.random() * 40) + 50}%`;
      | 		                     ^^^^^^^^^^^^^ Cannot call impure function
  656 | 	}, []);
  657 |
  658 | 	return (

(react-hooks/purity)